experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyTree.ts

```diff
@@ -516,34 +521,62 @@ function buildStructClass<TSchema extends StructSchema>(
 	const propertyDescriptorMap: PropertyDescriptorMap = {};
 	const ownPropertyMap: PropertyDescriptorMap = {};
 
-	for (const [key, field] of schema.structFields) {
+	for (const [key, fieldSchema] of schema.structFields) {
+		// Map containing setter implementations for field kinds that support setters.
+		const setters = new Map<FieldKind, (newContent: ContextuallyTypedNodeData) => void>([
```

Comment:
Considering this is likely a hot path for tree inserts, I think it's worth optimizing this a bit:
- can we avoid making this map and use a ternary instead?
- or, can we initialize it statically and re-bind the methods each time (or pass in the instance)?